src/Compilers/VisualBasic/Portable/Compilation/MethodCompiler.vb

```diff
@@ -850,10 +874,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
 
                 Dim boundBody = method.GetBoundMethodBody(compilationState, diagnosticsThisMethod)
 
-                If diagnosticsThisMethod.HasAnyErrors Then
-                    _diagnostics.AddRange(diagnosticsThisMethod)
-                    diagnosticsThisMethod.Free()
-                Else
+                If Not diagnosticsThisMethod.HasAnyErrors AndAlso DoEmitPhase Then
```

Comment:
Not sure if you did the swap up here as well, just making sure that Julien's below comment is seen here too.

> nit: consider swapping checks (I assume diagnosticsThisMethod.HasAnyErrors is more expensive). Possibly also below